aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIOProperties.java

```diff
@@ -50,6 +51,23 @@ public class S3FileIOProperties implements Serializable {
    */
   public static final String CLIENT_FACTORY = "s3.client-factory-impl";
 
+  /**
+   * This property is used to enable using the S3 Access Grants product to control authorization to
+   * S3 data. More information regarding this feature can be found at:
+   * https://aws.amazon.com/s3/features/access-grants/. The fallback-to-iam property allows users to
+   * customize whether or not they would like their jobs fall back to the Job Execution IAM role in
+   * case they get an Access Denied from the S3 Access Grants call. Further documentation regarding
+   * this flag can be found in the S3 Access Grants Plugin GitHub:
+   *
+   * <p>For more details, see: https://github.com/aws/aws-s3-accessgrants-plugin-java-v2
+   */
+  public static final String S3_ACCESS_GRANTS_ENABLED = "s3.access-grants.enabled";
+
+  public static final boolean S3_ACCESS_GRANTS_ENABLED_DEFAULT = false;
+  public static final String S3_ACCESS_GRANTS_FALLBACK_TO_IAM_ENABLED =
```

Comment:
can you add a doc explaining this feature flag?

Reply:
It is explained here: https://github.com/apache/iceberg/pull/9385/files#diff-4e5b26d66e1235fcac2c7e808f7800b757a08e668c1c3d9fad5ec0600de45e68R58-R60

Was there something else - or further explanation required?

Reply:
sorry I was not clear, I meant to move the portion of "The fallback-to-iam property allows users..." to be Java doc of this variable.

Reply:
Sure, done.